File: src/gui/cryoptix/src/features/strategy/strategy-state.ts
import { z } from "zod";

export const StrategyState = {
  Idle: 0,
  Starting: 1,
  Running: 2,
  Stopping: 3,
  Faulted: 4,
} as const;

export const StrategyStateSchema = z.enum(StrategyState);

export type StrategyState = z.infer<typeof StrategyStateSchema>;

File: src/api/Cryoptix.Strategy/State/StrategyState.cs
namespace Cryoptix.Strategy.State
{
    /// <summary>
    /// Defines the strategy state values.
    /// </summary>
    public enum StrategyState
    {
        /// <summary>
        /// Specifies the idle value.
        /// </summary>
        Idle,
        /// <summary>
        /// Specifies the starting value.
        /// </summary>
        Starting,
        /// <summary>
        /// Specifies the running value.
        /// </summary>
        Running,
        /// <summary>
        /// Specifies the stopping value.
        /// </summary>
        Stopping,
        /// <summary>
        /// Specifies the faulted value.
        /// </summary>
        Faulted,
    }
}
